### DISCUSSION TITLE
Game crashes when flight stick gets connected

### DISCUSSION THREAD

**Discussion by LetalexAlex** [ppy/osu#31136] [Q&A] [answered]
I recently bought a flight stick (precisely the [KROM FLIGHT](https://kromgaming.com/en/controllers/k-flight) stick).

Whenever I connect it to my pc and osu! is running, the game will just instantly crash without warning and terminate the process.
The game won't even open a window if the stick is connected prior to launching the game, but logs will still show up but get cut off almost instantly. 

Both logs show little to no information on what could have possibly happened except these two final lines in the input log:

```txt
[Tablet] HidSharpDeviceRootHub: Changes: 3, Add: 3, Remove: 0
[Tablet] RootHub: Invoking DevicesChanged
```
Input logs do not get generated in case 2.

Putting this in discussions since it's probably a personal setup problem and not critical since I can just unplug the controller to play, but it would be nice to see if it will be possible to have both the game and the controller on.

---

OS: Windows 11 23H2

LOGS:

[logs.zip](https://github.com/user-attachments/files/18142017/logs.zip) (game launched with stick unplugged and plugged later)

[logs_plugged.zip](https://github.com/user-attachments/files/18142021/logs_plugged.zip) (game launched with stick plugged)

**Comment by bdach** [comment]
Please attach Windows Event Viewer info by performing the following steps:
1. Type `Event viewer` in the Windows search bar (shortcut: <kbd>Windows</kbd>+<kbd>S</kbd>) and press <kbd>Enter</kbd>, or open the *Run...* button (<kbd>Windows</kbd>+<kbd>R</kbd>), type in `eventvwr` and press <kbd>Enter</kbd>.
2. The *Event viewer* window will appear. Expand *Windows Logs* on the left and choose *Application*. It might take a while to load.
3. In the *Actions* pane on the right click *Save All Events As...*, then choose a location to save to.
4. On the *Display Information* popup, select the second option, then the checkbox to show all available languages, then *English (UK)* (US would also do fine).
5. Append the resulting `.evtx` file to this issue.

**Comment by LetalexAlex** [comment, marked as answer]
Sorry for this really late answer, I was busy with vacations, I have figured out that the problem is that the stick's manager program is written for 32-bit systems for some reason. The stick is able to work without the program and I don't really need it so i uninstalled it. Game (and also other games other than osu! that kept crashing for the same reason) now work flawlessly. Thanks for the response and I will now close this discussion.
